packages/core/src/stream-loading/__tests__/tip-fetcher.test.ts

```diff
@@ -44,7 +44,10 @@ describe('TipFetcher test', () => {
       if (message.typ == MsgType.QUERY && message.stream.equals(streamID)) {
         const tipMap = new Map().set(streamID.toString(), tip)
         const response: PubsubMessage = { typ: MsgType.RESPONSE, id: message.id, tips: tipMap }
-        ipfs2.pubsub.publish(TOPIC, serialize(response))
+        ipfs2.pubsub.publish(TOPIC, serialize(response)).catch((error) => {
```

Comment:
This is more like a harness, so `void ipfs2.pubsub.publish` should be enough IMO.

Reply:
In the context of this test, ignoring a promise's result and potential errors with void can indeed lead to misleading test outcomes. If ipfs2.pubsub.publish() fails and throws an error, the promise is ignored due to void, the test might incorrectly pass because the error is never caught or handled.
Thus I added logic to catch the error and rethrow it to ensure the test fails in a scenario where this throws an error.

Reply:
Okay, going to treat it as a minor stylistic difference.

> the promise is ignored due to void

Result of an expression that goes into `void` gets replaced with `undefined`. The expression is still evaluated, i.e. `void foo()` executes `foo()` first. Here `ipfs.pubsub.publish` still gets executed and a Promise gets created. If the promise gets rejected, the fact it is ignored as a value does not affect its lifecycle: Node.js would still try to find  an error handler for it.

`promise.catch(e => throw e)` results in a `Promise`. `catch` clause re-throws an error. It is a no-op. Could be omitted.

So, both `void ipfs2.pubsub.publish` and `ipfs2.pubsub.publish.catch()` are equivalent. The difference is one suggests Promise is ignored, and another persuades the error is handled for real, while it is not. I'd prefer just the shorter version :)

Reply:
Hey, @samika98 mind removing `catch(() => throw)` clause please? It is a semantical no-op :)

Reply:
Added the fix, thanks for the suggestion :)